code/Model/EditableCustomRule.php

```diff
@@ -230,6 +231,63 @@ public function buildExpression()
         return $result;
     }
 
+
+    /**
+     * Determines whether the rule is satisfied, based on provided form data.
+     * Used for php validation of required conditional fields
+     *
+     * @param array $data Submitted form data
+     * @return boolean
+     * @throws LogicException Invalid ConditionOption is set for this rule.
+     */
+    public function validateAgainstFormData($data)
+    {
+
+        $controllingField = $this->ConditionField();
+
+        if (!isset($data[$controllingField->Name])) {
+            return false;
+        }
+
+        $valid = false;
+
+        $targetFieldValue = $this->FieldValue;
+        $actualFieldValue = $data[$controllingField->Name];
+
+        switch ($this->ConditionOption) {
+            case 'IsNotBlank':
+                $valid = ($actualFieldValue !== '');
+                break;
+            case 'IsBlank':
+                $valid = ($actualFieldValue === '');
+                break;
+            case 'HasValue':
+                $valid = ($actualFieldValue === $targetFieldValue);
```

Comment:
will using `===` sometimes mess up if a user enters a digit? i.e any chance of validation messing up because `1 === '1'` is not true?

Reply:
The values in `$data` are coming straight from the $_POST array which should always come in as strings no matter what. There's no effort anywhere to recast them to a specific type.

`$targetFieldValue` is also stored as a string.

Reply:
ok